### PR TITLE
Tlmblgis 135

### PR DESCRIPTION
**Pull request by gsilver**

### CODE REVIEW

File: mbofui/app/js/controllers.js [hunk 1]
```diff
@@ -296,6 +302,56 @@ ocellus.controller('mapController', ['$compile', '$scope', '$rootScope','$filter
...
+    data = {
+      'intention': intention,
+      'event': targetEvent,
+      'respondent': respondent
+    };
+    Bof.IntendPut(intentionUrl, data).then(function(result) {
+      leafletData.getMap().then(function(map) {
+        map.closePopup();
+      });
+      Bof.GetIntentions('/api/intentions/?username=self').then(function(intentionsList) {
```
Comment (dovek) [inline mbofui/app/js/controllers.js]: /api/intentions/?username=self is used 3 times in this same file. Can we make this a constant?

File: mbofui/app/js/controllers.js [hunk 2]
```diff
@@ -296,6 +302,56 @@ ocellus.controller('mapController', ['$compile', '$scope', '$rootScope','$filter
...
+      });
+    });
+  };
+  $scope.intendPost = function (intent, eventUrl) {
+    var eventId = _.last(_.compact(eventUrl.split('/')));
+    data = {
+        'intention': intent,
+        'event':eventUrl
+    };
+    var intentionsUrl = '/api/intentions/';
```
Comment (dovek) [inline mbofui/app/js/controllers.js]: I see the prefix is used here. Maybe we want to make this a constant as well. It's only used once though so not a necessity.

**Comment by dovek** [comment]
As previously discussed, comments would make this code a little easier to understand.

File: mbofui/app/js/directives.js [hunk 3]
```diff
@@ -1,10 +1,10 @@
 /*jshint strict:false */
-ocellus.directive('popup', ['$http', '$compile', function($http, $compile) {
+ocellus.directive('popup', ['$http', '$compile', 'Bof', function($http, $compile, Bof) {
```
Comment (gsilver) [inline mbofui/app/js/directives.js:2]: Take out Factory object.

**Pull request merged**